prefect_sqlalchemy/database.py

```diff
@@ -934,6 +937,3 @@ def __getstate__(self):
     def __setstate__(self, data: dict):
         """Upon loading back, restart the engine and results."""
         self.__dict__.update(data)
-        self._start_engine()  # block initialization doesn't get called here
-        self._start_exit_stack()
-        self._unique_results = {}
```

Comment:
These will be set to None:
```
        data.update({k: None for k in {"_engine", "_exit_stack", "_unique_results"}})
```

And when get_engine / get_connection is called, it'll be back to normal.